### DISCUSSION TITLE
Run ncurses programs from Textadept

### DISCUSSION THREAD

**Discussion by BrandonIrizarry** [orbitalquark/textadept#16] [Q&A]
Hi, 

(running x86_64 linux, Textadept 11.0)

I was wondering how I could use the 'textadept.run.run_commands' feature, along with 'os.spawn' and possibly 'events.connect(events.RUN_OUTPUT...' to run an ncurses binary in a _separate_ terminal (say xterm) window.  Normally, Textadept will run the binary inside a Message buffer, but I would (naturally) need an ncurses program to run inside its own terminal window.

**Comment by orbitalquark** [comment]
`textadept.run.run()` references `textadept.run.run_commands`, but spawns as a non-terminal process. You would have to manually invoke `os.spawn()` with an xterm and then your run command in it. The message buffer can send input to a `textadept.run.run()`-spawned process via stdin, but I don't think that would work for a full-blown ncurses program.
